### PR TITLE
Accept all original solution words

### PR DESCRIPTION
**Pull request by skedwards88**
As pointed out in https://github.com/skedwards88/crossjig/pull/22, a puzzle could be generated and then the word list could be updated to exclude one of the words in the puzzle before the puzzle was solved. This PR counts all of the original words in the puzzle as valid, even if they are no longer in the word list.

https://github.com/skedwards88/word_logic/pull/2 makes the corresponding changes to the core crosswordValidQ function. The PRs can merge in any order.

### CODE REVIEW

File: src/logic/getWordsFromGrid.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      `The number of columns and number of rows in the grid must be equal.`
+    );
+  }
+
+  const numColumnsPerRow = grid.map((row) => row.length);
+  if (Math.min(...numColumnsPerRow) != Math.max(...numColumnsPerRow)) {
+    throw new Error(`All of the rows in the grid must have the same length.`);
+  }
+
+  const transposedGrid = grid.map((_, index) => grid.map((row) => row[index]));
```
Comment (jorendorff) [inline src/logic/getWordsFromGrid.js:40]: It would be nice to have this idiom in a function somewhere, so that it's just `transpose(grid)` here.

**Review by jorendorff — COMMENTED** [review]
:+1: Totally makes sense.

File: src/logic/getWordsFromGrid.js (hunk 1 above)
Comment (skedwards88) [reply src/logic/getWordsFromGrid.js:40]: Good idea. I use this in a few different places across multiple games, so I'll open a separate PR to consolidate it into the the skedwards88/word_logic package.

Comment (skedwards88) [reply src/logic/getWordsFromGrid.js:40]: > so I'll open a separate PR 

(For future ref: https://github.com/skedwards88/word_logic/pull/4)

**Pull request merged**